stan/math/prim/meta/plain_type.hpp

```diff
@@ -70,7 +70,7 @@ struct plain_type<T, require_t<is_detected<T, internal::plain_type_check_t>>> {
  */
 template <typename T>
 struct plain_type<T,
-                  require_t<bool_constant<
```

Comment:
@t4c1 I had to add this so that it would use the default for kernel expressions. Or should we add a `PlainObject` to kernel expressions?

Reply:
> Or should we add a PlainObject to kernel expressions?

No. Adding a require for eigen types here is fine. If/when we do need a `plain_type` for kernel expressions we should add a specialization, not `PlainObject`.